### PR TITLE
Update TestingPhoenixServer using SharedResource

### PR DESCRIPTION
**Pull request by chenjian2664**
## Description
Update TestingPhoenixServer using SharedResource

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
```

### CODE REVIEW

**Review by kokosing — APPROVED** [review]
Nice!

File: plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/PhoenixQueryRunner.java [hunk 1]
```diff
@@ -134,7 +134,7 @@ public static void main(String[] args)
             throws Exception
     {
         DistributedQueryRunner queryRunner = createPhoenixQueryRunner(
-                TestingPhoenixServer.getInstance(),
+                TestingPhoenixServer.getInstance().get(),
```
Comment (kokosing) [inline plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/PhoenixQueryRunner.java:137]: You need to close this lease

**Review by kokosing — CHANGES_REQUESTED** [review]

File: plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/PhoenixQueryRunner.java [hunk 2]
```diff
@@ -133,11 +134,13 @@ private static Session createSession()
     public static void main(String[] args)
             throws Exception
     {
+        SharedResource.Lease<TestingPhoenixServer> instance = TestingPhoenixServer.getInstance();
```
Comment (kokosing) [inline plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/PhoenixQueryRunner.java:137]: I think you should use `try-with-resources` and instance should be closed just before ending `main` method

Comment (chenjian2664) [reply plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/PhoenixQueryRunner.java:137]: You are right

**Comment by findepi** [comment]
What's the benefit of this change?

**Comment by chenjian2664** [comment]
> What's the benefit of this change?

Improve code readability, users know to get a "Lease" from a "shared resource" instead of a new instance. Any concern from your side?

**Comment by findepi** [comment]
`io.trino.testing.SharedResource#returnLease` does reference counting and triggers shutdown.
this is good when resource can be re-creater relatively cheaply, but obviously may lead to a resource being created multiple times, perhaps once for each test class.

What's the `TestingPhoenixServer` startup time?

**Comment by chenjian2664** [comment]
For TestingPhoenixServer, returnLease is similar to the previous shutdown action, and the timing of the current call is also similar to afterClass as before. IMO there is no additional overhead for creating and closing resources compared to the previous implementation, correct me if I am wrong.

It is true that the creation of `TestingPhoenixServer` is time-consuming, we may be able to:

- Implement a function similar to closeAfterClass in `TestingPhoenixServer`, without actively returning Lease return `TestingPhoenixServer` instead.
- We can rewrite the Close method without actually closing the server, and then actively close the Lease when we want to close it.
- A bit of a hack way :), obtain the Lease somewhere in advance, and then actively close the Lease when we want to close it.

What do you think

**Comment by kokosing** [comment]
> `SharedResource#returnLease` does reference counting and triggers shutdown.

I am under impression that previous implementation did exactly the same.

**Pull request merged**